src/routes/getUserID.ts

```diff
@@ -1,37 +1,55 @@
 import {db} from '../databases/databases';
-import {Logger} from '../utils/logger';
 import {Request, Response} from 'express';
+import {UserID} from '../types/user.model';
 
-export async function getUserID(req: Request, res: Response) {
-    let userName = req.query.username as string;
-
-    if (userName == undefined || userName.length > 64 || userName.length < 3) {
-        //invalid request
-        res.sendStatus(400);
-        return;
-    }
-    
+function getFuzzyUserID(userName: String): Promise<[{userName: String, userID: UserID }]>  {
```

Comment:
```suggestion
function getFuzzyUserID(userName: String): Promise<{userName: String, userID: UserID }[]>  {
```

Same for the other one. You are telling it that the array is always length 1 otherwise which could be causing the other issue you mentioned

Reply:
Oh I must've missed that when it was complaining about needing the promise interface

Reply:
fixed in 33a45ce